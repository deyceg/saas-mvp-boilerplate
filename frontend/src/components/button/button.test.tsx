// hello.test.js, again
import '@testing-library/jest-dom';
import 'jest-styled-components';
import { render } from '@testing-library/react';
import pretty from 'pretty';

import { StyledButton } from './index';

import { dark } from '../../definitions/styled-components';

// let container = null;
// beforeEach(() => {
//   // setup a DOM element as a render target
//   container = document.createElement('div');
//   document.body.appendChild(container);
// });

// afterEach(() => {
//   // cleanup on exiting
//   unmountComponentAtNode(container);
//   container.remove();
//   container = null;
// });

it('should render a styled button', () => {
  const { container } = render(<StyledButton />);
  expect(pretty(container)).toMatchInlineSnapshot();
  expect(container).toHaveStyleRule('color', dark.colors.background);
  //   const el = (
  //     <StyledButton type="primary" htmlType="submit">
  //       Sign Up
  //     </StyledButton>
  //   );
  //   act(() => {
  //     render(el, container);
  //   });
  //   expect(
  //     pretty(container.innerHTML),
  //   ).toMatchInlineSnapshot(); /* ... gets filled automatically by jest ... */
});
