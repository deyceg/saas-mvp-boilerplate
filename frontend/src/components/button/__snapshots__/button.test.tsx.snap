// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a styled button 1`] = `"<button type=\\"button\\" class=\\"ant-btn ant-btn-default\\"></button>"`;
